examples-c/ethereum_evm_precompiles.c

```diff
@@ -0,0 +1,90 @@
+/** Constantine
+ *  Copyright (c) 2018-2019    Status Research & Development GmbH
+ *  Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+ *  Licensed and distributed under either of
+ *    * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+ *    * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+ *  at your option. This file may not be copied, modified, or distributed except according to those terms.
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <constantine.h>
+
+int from_hex(byte *dst, size_t dst_len, const char *hex_src, size_t src_len) {
+    // converts the given `hex_src` to a byte buffer. `buffer` must already be
+    // allocated to len(hex_src) // 2!
+    if (src_len % 2 != 0) return -1; // Length must be even
+    else if(dst_len * 2 != src_len) return -2; // dest length must be half src len
```

Comment:
It's fine to have `-1` being code for either length mismatch or destination length mismatch or the first character to be invalid.

It's just an example, and checking length or the first character is easy. Remembering the +3 is unnatural